lib/ansible/modules/cloud/azure/azure_rm_containerregistry_facts.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_containerregistry_facts
+version_added: "2.7"
+short_description: Get Registry facts.
+description:
+    - Get facts of Registry.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group to which the container registry belongs.
+        required: True
+    name:
+        description:
+            - The name of the container registry.
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+
+'''
+
+EXAMPLES = '''
+  - name: Get instance of Registry
+    azure_rm_containerregistry_facts:
+      resource_group: sampleresourcegroup
+      name: sampleregistry
+
+  - name: List instances of Registry
+    azure_rm_containerregistry_facts:
+      resource_group: sampleresourcegroup
+'''
+
+RETURN = '''
+registries:
+    description: A list of dictionaries containing facts for registries.
+    returned: always
+    type: complex
+    contains:
+        id:
+            description:
+                - The resource ID.
+            returned: always
+            type: str
+            sample: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.ContainerRegistry/registr
+                    ies/myRegistry"
+        name:
+            description:
+                - The name of the resource.
+            returned: always
+            type: str
+            sample: myRegistry
+        location:
+            description:
+                - The location of the resource. This cannot be changed after the resource is created.
+            returned: always
+            type: str
+            sample: westus
+        admin_user_enabled:
+            description:
+                - Is admin user enabled.
+            returned: always
+            type: bool
+            sample: yes
+        sku:
+            description:
+                - The SKU name of the container registry.
+            returned: always
+            type: str
+            sample: Classic
+        provisioning_state:
+            description:
+                - Provisioning state of the container registry
+            returned: always
+            type: str
+            sample: Succeeded
+        login_server:
```

Comment:
is there way to get login credentials? i want to login to my container registry to push some image in next step

Reply:
yes, in fact there is, but there is a separate api for that, also an api to regenerate credentials. do you think we should include credentials here?

Reply:
Just checked azure_rm_containerregistry module, there's no way to set username/password there. So if not returned in facts, user  has no way to push image to  the registry if provisioned by ansible.  If you could provide interface in main module to let user set username/password, it would be better, then maybe we don't need username/password returned in facts.

Reply:
just followed exactly the same way of returning it as in container registry module